src/libraries/Common/src/System/Security/Cryptography/ManagedCompositeMLDsaComposer.cs

```diff
@@ -0,0 +1,597 @@
+
+                if (signature.AsSpan().TryCopyTo(destination))
+                {
+                    bytesWritten = signature.Length;
+                    return true;
+                }
+
+                bytesWritten = 0;
+                CryptographicOperations.ZeroMemory(destination);
+                return false;
+#endif
+            }
+
+            internal override bool VerifyData(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signature)
+            {
+#if NET
+                return _rsa.VerifyData(data, signature, _hashAlgorithmName, _padding);
+#else
+                // TODO data (msg representative) can be passed as an array (caller of VerifyData can allocate instead of rent)
+
+                // Composite ML-DSA virtual methods only accept ROS<byte> so we need to use ToArray() for signature
+                return _rsa.VerifyData(data.ToArray(), signature.ToArray(), _hashAlgorithmName, _padding);
+#endif
+            }
+
+            public static RsaComponent GenerateKey(CompositeMLDsaAlgorithm.RsaAlgorithm algorithm) => throw new NotImplementedException();
+            public static RsaComponent ImportPrivateKey(CompositeMLDsaAlgorithm.RsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+            {
+                Debug.Assert(IsAlgorithmSupported(algorithm));
+#if NETFRAMEWORK || NETSTANDARD2_0
+                // TODO netfx only has ImportParameters, so we need to implement the conversion from raw key to parameters
+                throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(RSA)));
+#else
+                Debug.Assert(!RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")));
+
+                RSA? rsa = null;
+
+                try
+                {
+                    rsa = RSA.Create();
+                    rsa.ImportRSAPrivateKey(source, out int bytesRead);
+
+                    if (bytesRead != source.Length)
+                    {
+                        // TODO resx
+                        throw new CryptographicException();
+                    }
+                }
+                catch (CryptographicException)
+                {
+                    rsa?.Dispose();
+                    throw;
+                }
+
+                return new RsaComponent(rsa, algorithm.HashAlgorithmName, algorithm.Padding);
+#endif
+            }
+
+            public static RsaComponent ImportPublicKey(CompositeMLDsaAlgorithm.RsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+            {
+                Debug.Assert(IsAlgorithmSupported(algorithm));
+
+#if NETFRAMEWORK || NETSTANDARD2_0
+                // TODO netfx only has ImportParameters, so we need to implement the conversion from raw key to parameters
+                throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(RSA)));
+#else
+                // TODO move browser to different file
```

Comment:
Browser should just be the NotImplemented.cs

Reply:
This file needs to be in the compilation for M.B.C.